config/authentication.go

```diff
@@ -0,0 +1,47 @@
+package config
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/kiali/kiali/config/security"
+	"github.com/kiali/kiali/log"
+)
+
+func Authentication(next http.Handler, credentials security.Credentials) http.Handler {
```

Comment:
I've read a couple of times the PR and I guess if this function could be named "AuthenticationHandler" it would help to understand it better (at least from me).
Not a blocker, but just a comment, this function is a wrapper to proxy and check authentication first.